lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -71,6 +71,7 @@ def serializable_hash_for_single_resource(options)
         end
 
         def resource_identifier_type_for(serializer)
+          return serializer.class._type unless serializer.class._type.nil?
```

Comment:
Might be nice to use `class_attribute :_type, instance_writer: false` so we can just `_type` here